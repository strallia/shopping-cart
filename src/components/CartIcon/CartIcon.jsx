import PropTypes from "prop-types";
import CartImage from "../../assets/cart-icon.png";
import styles from "./CartIcon.module.css";

const CartIcon = ({ totalItemsCount }) => {
  return (
    <div className={styles.container}>
      <img src={CartImage} className={styles.cart} />
      <div className={styles.count} aria-label="cart total" role="img">
        {totalItemsCount}
      </div>
    </div>
  );
};

CartIcon.propTypes = {
  totalItemsCount: PropTypes.number,
};

export default CartIcon;
