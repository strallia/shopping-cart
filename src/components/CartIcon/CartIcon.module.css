.container {
  position: relative;
  border: none;
  background-color: transparent;
  cursor: pointer;
}

.cart {
  max-width: 35px;
}

.count {
  position: absolute;
  min-width: 15px;
  top: 5px;
  left: 13px;
  font-size: 10px;
  background-color: var(--red);
  padding: 3px;
  border-radius: 50%;
  color: white;
  font-weight: bold;
  display: flex;
  justify-content: center;
}
