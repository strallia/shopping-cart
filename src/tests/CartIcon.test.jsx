import { render, screen } from "@testing-library/react";
import CartIcon from "../components/CartIcon/CartIcon";

describe("CartIcon component", () => {
  it("renders correctly with cart image and count tracker", () => {
    const { container } = render(<CartIcon />);
    expect(container).toMatchSnapshot();
  });

  it("the count tracker renders 0 by default", () => {
    render(<CartIcon />);
    expect(screen.getByRole("img", { name: "cart total" }));
  });

  it("for a cart with 2 total items, it renders 2 in the count tracker", () => {
    render(<CartIcon totalItemsCount={2} />);
    expect(screen.getByRole("img", { name: "cart total" }).textContent).toMatch(
      2
    );
  });

  it("for a cart with 1 total items, it renders 1 in the count tracker", () => {
    render(<CartIcon totalItemsCount={1} />);
    expect(screen.getByRole("img", { name: "cart total" }).textContent).toMatch(
      1
    );
  });
});
