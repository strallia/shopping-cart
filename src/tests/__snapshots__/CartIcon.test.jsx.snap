// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`CartIcon component > renders correctly with cart image and count tracker 1`] = `
<div>
  <div
    class="_container_3cc22e"
  >
    <img
      class="_cart_3cc22e"
      src="/src/assets/cart-icon.png"
    />
    <div
      aria-label="cart total"
      class="_count_3cc22e"
      role="img"
    />
  </div>
</div>
`;
